### PR TITLE
Bug 1605454 - Webrender experiment for Fenix

### PR DESCRIPTION
**Pull request by ktaeleman**
More information about this experiment here:
https://experimenter.services.mozilla.com/experiments/control-webrender-rollout-for-fenix/

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
-> Experiment, no test needed
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
-> Experiment, no screenshot needed
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features
-> Experiment, no user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/geckoBeta/java/org/mozilla/fenix/engine/GeckoProvider.kt [hunk 1]
```diff
@@ -39,6 +40,13 @@ object GeckoProvider {
             .debugLogging(Config.channel.isDebug)
             .build()
 
+        Experiments.withExperiment("webrender-performance-comparisson-experiment") {
```
Comment (mythmon) [inline app/src/geckoBeta/java/org/mozilla/fenix/engine/GeckoProvider.kt]: Typo: "comparisson" should be "comparison"

**Comment by ktaeleman** [comment]
Corresponding changes on geckoview side:
https://phabricator.services.mozilla.com/D58568

**Comment by liuche** [comment]
I'm talking with Kris later today, about the Experiments + Fenix process.

**Comment by sblatz** [comment]
@liuche Since this has an approval I'm going to put a `do not land` on it until you hear back 😄

**Comment by liuche** [comment]
@ekager after the Experiments meeting 1/9, if everything seems good to run this experiment on Nightly, please land this!

**Comment by ekager** [comment]
Currently this code is calling the experiment branch code before `Experiments.initaialize` (even though the engine is lazy we start to warm it up almost immediately in `FenixApplication`)

@mythmon can you confirm we'll need to switch around some calls here? (We may have to initialize experiments earlier)

**Comment by mythmon** [comment]
In that case, we'll need to re-order the calls. Ideally we'd start initializing experiments as soon as possible. Anything that we do after that can't be experimented on reliably.

**Comment by ekager** [comment]
My first instinct is to just move the `Experiments.initialize` to the top of `setupInMainProcessOnly` but I'm checking @mcomella to make sure thats okay perf wise

**Comment by ekager** [comment]
@mcomella will monitor for any perf regressions, but has approved the new location of experiments initialization (before engine init)

**Comment by mcomella** [comment]
Perf note: I checked the latest ND results and I did not notice any regression in cold startup performance (note: we went from 1.69s to 1.77s but we've been seeing intermittent timing errors). In particular, `Experiments.initialize` did not appear in the ND profiles at all (not that I trust them to show all the methods all that much).

**Comment by ktaeleman** [comment]
Thanks everyone!

**Pull request merged**